### ISSUE TITLE
Not showing top-level asynchronous function defintion with jedi.api.names

### ISSUE THREAD

**Issue by CXuesong** [davidhalter/jedi#1309]
I tried to use `jedi.api.names` to generate document outline for the following code
```python
# AsyncFunctions.py
import asyncio

def func() -> int:
    asyncio.run(func_async(0))

async def func_async(arg) -> int:
    await asyncio.sleep(1000)
    return 100
```
with the following code (flattened document outline)
```python
import io
import jedi

with io.open("AsyncFunctions.py", "r") as f:
    content = f.read()
    for n in jedi.api.names(content, f.name, all_scopes=True):
        print(n, n.parent(), n.is_definition())
```

It gives me the following output
```
<Definition module asyncio> <Definition module AsyncFunctions> True
<Definition def func> <Definition module AsyncFunctions> True
<Definition def func_async> <Definition module AsyncFunctions> True
<Definition param arg> <Definition def func_async> True
```
The result here is satisfactory.

However, if I remove `all_scopes=True` (use `False` by default), the output becomes
```
<Definition module asyncio> <Definition module AsyncFunctions> True
<Definition def func> <Definition module AsyncFunctions> True
```

The problem here, is `func_async`, regardless of it being a top-level function, is missing. Is this a bug or by design?

**Comment by davidhalter** [comment]
It's probably a bug :)

**Comment by CXuesong** [comment]
Okay. It seems that I need to make some modifications to [`get_module_names`](https://github.com/davidhalter/jedi/blob/8d0c4d3cec4973a2722652bf33f093ac3f1f385a/jedi/evaluate/helpers.py#L164), because async functions are actually functions wrapped in [`async_stmt`](https://github.com/davidhalter/parso/blob/1e25445176185f8ce27460cfa7162c897dd7527b/parso/python/grammar38.txt#L71).

I also noted there are still top-level function parameters and in-function local variables included in the result even if `all_scopes=False`, i.e. #1202, and curious about the current situation of that issue and PR #1203 . May I try to fix this as well?

**Comment by davidhalter** [comment]
I just merged #1203, so feel free to work on it :)

**Comment by davidhalter** [comment]
#1313 was merged and should solve this issue, thanks for helping out!
